<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ModbusSlave</class>
 <widget class="QMainWindow" name="ModbusSlave">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>898</width>
    <height>855</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="SerialPort">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>20</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>3ds</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Serial Port</string>
    </property>
   </widget>
   <widget class="QLabel" name="CoilsQuantuty">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>180</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>3ds</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Coils Quantity</string>
    </property>
   </widget>
   <widget class="QPushButton" name="run_button">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>20</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>3ds</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Run</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="coils_address_input">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>130</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>16</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="placeholderText">
     <string>end with enter</string>
    </property>
   </widget>
   <widget class="QLabel" name="Function">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>70</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>3ds</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Function </string>
    </property>
   </widget>
   <widget class="QLineEdit" name="coils_quantity_input">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>180</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>16</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="placeholderText">
     <string>end with enter</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="slave_id_input">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>70</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>16</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="placeholderText">
     <string>end with enter</string>
    </property>
   </widget>
   <widget class="QComboBox" name="function_code">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>70</y>
      <width>211</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>16</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QPushButton" name="connect_button">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>20</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>3ds</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Connect</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="serial_port_input">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>20</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>16</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="placeholderText">
     <string>end with enter</string>
    </property>
   </widget>
   <widget class="QLabel" name="CoilsAddress">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>130</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>3ds</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Coils Address</string>
    </property>
   </widget>
   <widget class="QLabel" name="SlaveID">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>70</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>3ds</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Slave ID</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="CoilsBlock">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>320</y>
      <width>341</width>
      <height>491</height>
     </rect>
    </property>
   </widget>
   <widget class="QTableWidget" name="HoldingRegister">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>320</y>
      <width>341</width>
      <height>491</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="RgisterAddress">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>130</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>3ds</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Register Address</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="register_address_input">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>130</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>16</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="placeholderText">
     <string>end with enter</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="register_quantity_input">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>180</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Times New Roman</family>
      <pointsize>16</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="placeholderText">
     <string>end with enter</string>
    </property>
   </widget>
   <widget class="QLabel" name="RegisterQuantuty">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>180</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>3ds</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Register Quantity</string>
    </property>
   </widget>
   <widget class="QPushButton" name="stop_button">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>20</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>3ds</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Stop</string>
    </property>
   </widget>
   <widget class="QLabel" name="Register">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>280</y>
      <width>211</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>18</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Holding Register</string>
    </property>
   </widget>
   <widget class="QLabel" name="Coils">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>280</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>18</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Coils</string>
    </property>
   </widget>
   <widget class="QLabel" name="Running">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>250</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>18</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="text">
     <string>Running</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="isRunning">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>230</y>
      <width>81</width>
      <height>71</height>
     </rect>
    </property>
    <property name="digitCount">
     <number>1</number>
    </property>
   </widget>
   <widget class="QLabel" name="SlaveID_2">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>70</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>3ds</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>default 1</string>
    </property>
   </widget>
   <widget class="QLabel" name="CoilsAddress_2">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>130</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>3ds</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>default 0</string>
    </property>
   </widget>
   <widget class="QLabel" name="CoilsQuantuty_2">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>180</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>3ds</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>default 10</string>
    </property>
   </widget>
   <widget class="QLabel" name="RgisterAddress_2">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>130</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>3ds</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>default 10</string>
    </property>
   </widget>
   <widget class="QLabel" name="RegisterQuantuty_2">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>180</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>3ds</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>default 10</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>898</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuModbusSlave">
    <property name="title">
     <string>ModbusSlave</string>
    </property>
   </widget>
   <addaction name="menuModbusSlave"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
